libs/extensions/src/TTAlpsSelections.cpp

```diff
@@ -34,24 +43,47 @@ bool TTAlpsSelections::PassesLooseSemileptonicSelections(const shared_ptr<Event>
   return true;
 }
 
-bool TTAlpsSelections::PassesSingleLeptonSelections(const shared_ptr<Event> event) {
+bool TTAlpsSelections::PassesSignalLikeSelections(const shared_ptr<Event> event, shared_ptr<CutFlowManager> cutFlowManager) {
   float metPt = event->Get("MET_pt");
-  if (metPt <= 30) return false;
+  if (!inRange(metPt, eventSelections["MET_pt"])) return false;
 
   AddExtraCollections(event);
 
-  if (event->GetCollectionSize("LeptonPt30") != 1) return false;
+  if (!inRange(event->GetCollectionSize("GoodLeptons"), eventSelections["nGoodLeptons"])) return false;
+  if (!inRange(event->GetCollectionSize("GoodBtaggedJets"), eventSelections["nGoodBtaggedJets"])) return false;
+  if (!inRange(event->GetCollectionSize("GoodJets"), eventSelections["nGoodJets"])) return false;
 
-  int nLeptons15 = event->GetCollectionSize("LeptonPt15");
-  if (nLeptons15 > 1) return false;
-  if (nLeptons15 == 1) {
-    auto leadingLepton = event->GetCollection("LeptonPt30")->at(0);
-    auto survivingLepton = event->GetCollection("LeptonPt15")->at(0);
-    if (survivingLepton != leadingLepton) return false;
+  auto goodLeptons = event->GetCollection("GoodLeptons");
+  int requiredGoodMuons = 3;
+  for (auto lepton : *goodLeptons) {
+    if (lepton->GetOriginalCollection() != "Electron") continue;
+    requiredGoodMuons = 2;
+    break;
   }
+  if (event->GetCollectionSize("GoodMuons") < requiredGoodMuons) return false;
+  cutFlowManager->UpdateCutFlow("6_twoAdditionalMuons");
```

Comment:
What is the number 6 here? Is it for the order of the cuts? Could we make the numbering automatic?

Reply:
Yes, I was thinking of making it automatic and then I created PR before actually doing it ;) I pushed a commit now which implements that.